Raspberry Pico 2 porting (#14831)
apache/nuttx: Documentation/platforms/arm/rp23xx/index.rst

2. Download NuttX and the companion applications.  These must both be
   contained in the same directory::

    git clone https://github.com/apache/nuttx.git nuttx
    git clone https://github.com/apache/nuttx-apps.git apps

Building NuttX
==============

1. Change to NuttX directory::

    cd nuttx

2. Select a configuration. The available configurations
   can be listed with the command::

    ./tools/configure.sh -L

3. Load the selected configuration.::

    make distclean
    ./tools/configure.sh raspberrypi-pico-2:usbnsh

4. Modify the configuration as needed (optional)::

    make menuconfig

5. Build NuttX::

    make

Flash boot
==========

By default, the system is built to build and run from the flash
using XIP. By using the default `BOOT_RUNFROMFLASH` configuration,
the full image is run from the flash making most of the internal
SRAM available for the OS and applications, however the execution
is slower. The cache can speed up, but you might want set your
time critical functions to be placed in the SRAM (copied from
the flash on startup).

It is also possible to execute from SRAM, which reduces the
available SRAM to the OS and applications, however it is very

> useful when debugging as erasings and rewriting the flash on

Reviewer: ```suggestion
useful when debugging as erasing and rewriting the flash on
```

every build is tedious and slow. This option is enabled with
`BOOT_RUNFROMISRAM` and requires `openocd`` and/or `gdb`.

There is a third option which is to write the firmware on the
flash and it gets copied to the SRAM. This is enabled with
`CONFIG_BOOT_COPYTORAM` and might be useful for time critical

> applications, on the expense of reduced useable interna SRAM

Reviewer: ```suggestion
applications, on the expense of reduced usable internal SRAM
```

The internal SRAM is used as the main heap for kernel and
applications, as if there was no PSRAM configured. The
external PSRAM is configured as a separate user heap called
`psram` and can be used through the global variable
`g_psramheap` after including `rp23xx_heaps.h`

Programming
============

Programming using BOOTSEL
-------------------------

Connect  board to USB port while holding BOOTSEL.
The board will be detected as USB Mass Storage Device.
Then copy "nuttx.uf2" into the device.
(Same manner as the standard Pico SDK applications installation.)

Programming with picotool
-------------------------

You can use picotool to load the elf (or the uf2)::

    picotool load nuttx -t elf

Programming using SWD debugger
------------------------------

Most boards provide a serial (SWD) debug port.
The "nuttx" ELF file can be uploaded with an appropriate SDB programmer
module and companion software (openocd and gdb)

Running NuttX
=============

Most builds provide access to the console via UART0.  To access this
GPIO 0 and 1 pins must be connected to the device such as USB-serial converter.

The `usbnsh` configuration provides the console access by USB CDC/ACM serial

> devcice.  The console is available by using a terminal software on the USB host.

Reviewer: ```suggestion
device.  The console is available by using a terminal software on the USB host.
```
